src/MCPClient/lib/metrics.py

```diff
@@ -0,0 +1,234 @@
+    ["transfer_type", "failure_type"],
+)
+transfer_error_timestamp = Gauge(
+    "mcpclient_transfer_error_timestamp",
+    "Timestamp of most recent transfer failure",
+    ["transfer_type", "failure_type"],
+)
+
+sip_started_counter = Counter("mcpclient_sip_started_total", "Number of SIPs started")
+sip_started_timestamp = Gauge(
+    "mcpclient_sip_started_timestamp", "Timestamp of most recent SIP started"
+)
+sip_error_counter = Counter(
+    "mcpclient_sip_error_total", "Number of SIP failures", ["failure_type"]
+)
+sip_error_timestamp = Gauge(
+    "mcpclient_sip_error_timestamp",
+    "Timestamp of most recent SIP failure",
+    ["failure_type"],
+)
+
+aips_stored_counter = Counter("mcpclient_aips_stored_total", "Number of AIPs stored")
+dips_stored_counter = Counter("mcpclient_dips_stored_total", "Number of DIPs stored")
+aips_stored_timestamp = Gauge(
+    "mcpclient_aips_stored_timestamp", "Timestamp of most recent AIP stored"
+)
+dips_stored_timestamp = Gauge(
+    "mcpclient_dips_stored_timestamp", "Timestamp of most recent DIP stored"
+)
+aip_processing_time_summary = Summary(
+    "mcpclient_aip_processing_seconds",
+    "AIP processing time, from first file recorded in DB to storage in SS",
+)
+dip_processing_time_summary = Summary(
+    "mcpclient_dip_processing_seconds",
+    "DIP processing time, from first file recorded in DB to storage in SS",
+)
+aip_files_stored_counter = Counter(
+    "mcpclient_aip_files_stored_total", "Number of files stored in AIPs"
+)
+dip_files_stored_counter = Counter(
+    "mcpclient_dip_files_stored_total", "Number of files stored in DIPs"
+)
+aip_size_counter = Counter("mcpclient_aip_size_bytes", "Number of bytes stored in AIPs")
+dip_size_counter = Counter("mcpclient_dip_size_bytes", "Number of bytes stored in DIPs")
+
+archivematica_info = Info("archivematica_version", "Archivematica version info")
+environment_info = Info("environment_variables", "Environment Variables")
+
+
+def skip_if_prometheus_disabled(func):
+    @functools.wraps(func)
+    def wrapper(*args, **kwds):
+        if settings.PROMETHEUS_ENABLED:
+            return func(*args, **kwds)
+        return None
+
+    return wrapper
+
+
+def init_counter_labels():
+    # Zero our counters to start, by intializing all labels. Non-zero starting points
+    # cause problems when measuring rates.
+    modules_config = ConfigParser.RawConfigParser()
+    modules_config.read(settings.CLIENT_MODULES_FILE)
+    for script_name, _ in modules_config.items("supportedBatchCommands"):
+        job_counter.labels(script_name=script_name)
+        job_processed_timestamp.labels(script_name=script_name)
+        job_error_counter.labels(script_name=script_name)
+        job_error_timestamp.labels(script_name=script_name)
+        task_execution_time_summary.labels(script_name=script_name)
+
+    failure_types = ("fail", "reject")
+
+    for transfer_type in ("Standard", "Dataverse", "Dspace", "TRIM", "Maildir"):
```

Comment:
Is there any technical reason to leave bagged transfers out?

Reply:
No. That list is taken from the values passed to the `archivematicaSetTransferType_v0.0` client script (in workflow.json) — I couldn't find anything more definitive. Is bagged transfer another type in that sense?

Reply:
Ah! I need to look but if I remember correctly bagged transfers start on a different approval workflow (or starting point). Maybe they don't pass through `archivematicaSetTransferType_v0.0`.

Reply:
I don't think that's a type in that sense? I'm going to leave as is for now (it's just initialization to zero) and if "Bagged" starts showing up somewhere, we can add it.